packages/@lwc/ssr-compiler/src/compile-js/generate-markup.ts

```diff
@@ -98,6 +104,11 @@ export function addGenerateMarkupExport(
         program.body.unshift(bImportDefaultDeclaration('tmpl', defaultTmplPath));
     }
 
+    let connectWireAdapterCode: Statement[] = [];
+    if (state.wireAdapters.length) {
+        connectWireAdapterCode = bWireAdaptersPlumbing(state.wireAdapters);
+    }
```

Comment:
If no wire adapters are detected on the component, we don't bother injecting the wire-related code.